### ISSUE TITLE
LoRA training freezing after one epoch finishes

### ISSUE THREAD

**Issue by dareaker** [bmaltais/kohya_ss#1178]
Hi there,

This is my first time using Kohya, but I've been fiddling with it for 3 days now. This one problem keeps coming up over and over and over again and all of my raking the internet and this forum pulls up nobody with the same issue.

My training is freezing after the first epoch finishes. But it also appears to be the last epoch?

![Screenshot 2023-07-12 082820](https://github.com/bmaltais/kohya_ss/assets/139360907/4d134016-174b-42f7-a4bd-f4c21a737db2)
![Screenshot 2023-07-12 082750](https://github.com/bmaltais/kohya_ss/assets/139360907/8f5be5c6-2b95-4c28-8631-92fbaec27308)

No specific errors (except the triton one I've heard I can ignore). I've got the default 3 epochs for my collection of 153 pics for training a character LoRA. I'm running 8gb of VRAM on my NIVIDIA GeForce RTX 3070 Ti Laptop and plenty of memory, but the weird thing is my VRAM goes silent as soon as that last/first epoch finishes. I can watch the memory on my card stop being used as soon as we hit that fateful 100% epoch.

I have samples set up too, and can see the loRA trying to do its thing. It's infuriating, because they look SO GOOD for early samples, but I can't get them to progress any further than this spot.

Please help! Am I just dumb? lmfao

**Comment by james-things** [comment]
Did you just install/update?

I am not sure if it is the same issue, but on the most recent commit, I crash while trying to generate my second sample image after an epoch completes, with an error related to precision mismatch (float vs fp16 conflict).

I was able to get training working again by reverting to a commit from the prior day, however I am using traditional Dreambooth and again am getting an error where you are not. Still, the crash between epochs makes me wonder if the issues are related.

To revert to the prior commit, I used this command. Be aware you will lose *any and all* modifications you have made to the local copy of the repo by running this command:

`git reset --hard 1ba606c`

**Comment by dareaker** [comment]
It is a pretty recent install, yes. Although it was installed last week. Here's what I've tried:

-  I tried rolling it back, no dice. 
- Just tried updating it to the most recent pull of 17hrs ago. Update went fine, issue is still happening
- I've tried both torches
- Uninstalling/reinstalling twice, not counting the rolling back and then updating
- Bouncing between AdamW8bit and AdamW
- Changing the mixed precision and fixed precision

And so many other things I can't remember at the moment.

But yeah, TLDR: Tried it, am back up to date, and still having the exact same issue

**Comment by james-things** [comment]
Sorry to have lead you in the wrong direction then! Re-reading your post, it really is quite difficult to figure out what might be going on, especially since you are not running up against VRAM limitations.

The only other thing that stuck out to me was that you mentioned you are using a laptop. I wonder if there is any possibility that somewhere along the line, the proper GPU isn't being selected, since the configuration is a little different in a laptop?

I am not too sure how you might explore this possibility though. I saw that the manual accelerate configuration allows you to specify a particular GPU, but I have not taken the time to fully understand the role of accelerate in the overall application, to be totally honest, so I am not sure if that is worth checking out or not.

**Comment by dareaker** [comment]
I'm glad it's not just me then LOL I can't find any errors?? If there was an error that'd be something to work with. But just... freezing? Ugh.

Ahhh that's one of the things I tried and couldn't think of. A good thought, because I'm pretty sure it _was_ using GPU incorrectly in the beginning. However, I did manually set accelerate to use my NIVIDIA gpu, which has the id 0. And I also selected for it to only use 1 gpu and never use cpu. This does seem to be working, in that the internal gpu and cpu are barely bumping when I'm running the script. But the problem remained the same before and after the fix. No changes in behavior, although I did watch the  load all shift to my card, which sped up the parts of the process that work.

Thank you for all your thoughts and suggestions! At the very least it's validating that this is in fact a weird problem and I'm not just a dummy LOL
